clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -331,6 +336,23 @@ public synchronized void removeListener(Listener listener) {
         this.listeners.remove(listener);
     }
 
+    /**
+     * "Poison" the metadata to indicate that metadata updates are no longer possible. This is typically used when the
+     * thread responsible for performing metadata updates is exiting and needs a way to relay this information to any
+     * other thread(s) that could potentially wait on metadata update to come through.
+     */
+    public synchronized void poison() {
+        this.version = -1;
```

Comment:
I thought it might be a bit awkward to have a `close()` method for a class like `Metadata` that has no underlying resources really. That said, I also introduced `close()` for `MetadataUpdater` so may be we should do the same for `Metadata` as well. I will update the PR.